src/SwipeProfiles.js

```diff
@@ -0,0 +1,193 @@
+import React from 'react';
+import './SwipeProfile.css';
+import {withRouter, Link} from 'react-router-dom'
+
+class SwipeProfiles extends React.Component {
+
+    constructor(props){
+        super(props);
+        this.state = {
+            age: '',
+            firstName: '',
+            description: '',
+            interests: '',
+            gender: '',
+            workplace: '',
+            potentialFriends: [],
+            displayedUserId: '',
+            error: '',
+        };
+
+        this.getPotentialFriendList = this.getPotentialFriendList.bind(this);
+        this.displayProfile = this.displayProfile.bind(this);
+        this.handleSwipe = this.handleSwipe.bind(this);
+    }
+
+    getPotentialFriendList() {
+      const id = this.props?.location?.state?.id;
+      var formData = new FormData();
+      formData.append("userId", id)
+      const myRequest = new Request('http://127.0.0.1:5000/getPotentialFriends', {
+            method: 'POST',
+            body: formData,
+          });
+      fetch(myRequest)
+        .then(res => res.json())
+        .then(res => { 
+          if(res.response === "Success"){
+            var potentialFriendsList = res.potentialListId;
+            const displayProfileId = potentialFriendsList.pop();
+            console.log(displayProfileId);
+            this.setState({
+              potentialFriends: potentialFriendsList,
+              displayedUserId: displayProfileId,
+              error: '',
+            });
+
+            this.displayProfile();
+          }        
+          else{      
+            this.setState({
+              error: res.response
+            });
+        }
+        })
+        .catch((error) => {
+            this.setState({
+              error: "Error connecting to backend"
+            });
+          }
+        );
+
+    }
+
+
+
+
+    displayProfile(){
+        const displayId = this.state.displayedUserId;
+        if(displayId) {
+        var formData = new FormData();
+        formData.append("userId", displayId);
+        const myRequest = new Request('http://127.0.0.1:5000/displayProfile', {
+              method: 'POST',
+              body: formData,
+            });
+        fetch(myRequest)
```

Comment:
If we are fetching from database for potential users every time the user swipe, and only using one potential user id before the next swipe, why do we have to make the api call to grab related info to potential user again here? Could we just get this info when we went to get the potential user id, since we only care about 1 potential user id at a time?